src/Tracking/TrackingServiceProvider.php

```diff
@@ -31,19 +31,22 @@ public function register() {
 	 */
 	public function boot() {
 		if ( Track::isTrackingEnabled() ) {
-			$this->registerTrackEvents();
-			Hooks::addAction( 'shutdown', TrackJobScheduler::class, 'schedule', 999 );
 			Hooks::addAction( TrackJobScheduler::CRON_JOB_HOOK_NAME, TrackJob::class, 'send' );
 		}
 
-		if ( is_admin() && Track::checkEnvironment() ) {
-			Hooks::addFilter( 'give_get_settings_advanced', AdminSettings::class, 'addSettings' );
-			Hooks::addAction( 'give_opt_in_into_tracking', AdminActionHandler::class, 'optInToUsageTracking' );
-			Hooks::addAction( 'give_hide_opt_in_notice_shortly', AdminActionHandler::class, 'optOutFromUsageTracking' );
-			Hooks::addAction( 'give_hide_opt_in_notice_permanently', AdminActionHandler::class, 'optOutFromUsageTracking' );
-			Hooks::addAction( 'update_option_give_settings', AdminActionHandler::class, 'optInToUsageTrackingAdminGrantManually', 10, 2 );
-			Hooks::addAction( 'give_setup_page_before_sections', UsageTrackingOnBoarding::class, 'addNotice', 0 );
-			Hooks::addAction( 'admin_notices', UsageTrackingOnBoarding::class, 'addNotice' );
+		if ( is_admin() ) {
+			$this->registerTrackEvents();
```

Comment:
@ravinderk is there a reason to register tracking events if tracking is not enabled?

Reply:
@alaca Good catch. I fixed it: https://github.com/impress-org/givewp/compare/09dc99f0c3c1...cf9e57c670eb